Standards/scs-XXXX-v1-volume-backup-service.md

```diff
@@ -0,0 +1,71 @@
+
+## Motivation
+
+The [volume backup functionality of OpenStack](https://docs.openstack.org/cinder/latest/admin/volume-backups.html) is a feature that is not available in all OpenStack clouds per default.
+The feature requires a backend to prepared and configured correctly before it can be used.
+In Cinder this is a separate configuration to the general storage backend of the volume service and is not mandatory.
+Thus, an arbitrary OpenStack cloud may or may not offer this feature.
+
+This standard aims to make this functionality the default in SCS clouds so that customers can expect the feature to be usable.
+
+## Design Considerations
+
+The standard should make sure that the feature is available and usable but should not limit the exact implementation (e.g. choice of backend driver) any further than necessary.
+
+### Options considered
+
+#### Only recommend volume backup feature, use images as alternative
+
+As an alternative to the volume backup feature of the Block Storage API, Glance images can also be created based on volumes and act as a backup under certain circumstances.
+As an option, this standard could keep the actual integration of the volume backup feature optional and guide users how to use images as backup targets instead in case the feature is unavailable.
+
+However, it is not guaranteed that the Glance backend storage is separate from the volume storage.
+For instance, both could be using the same Ceph cluster.
+In such case, the images would not count as genuine backups.
+
+Although users are able to download images and transfer them to a different storage location, this approach might also prove unfeasible depending on the image size and the existence (or lack) of appropriate target storage on the user side.
+
+Furthermore, incremental backups are not possible when creating Glance images from volumes either.
+This results in time-consuming backup operations of fully copying a volume everytime a backup is created.
+
+#### Mandate the availability of the volume backup feature
+
+This option is pretty straightforward.
+It would make the volume backup feature mandatory for SCS clouds.
+This way users can expect the feature to be available and usable.
+
+With this, users can leverage functionalities like incremental backups and benefit from optimized performance of the backup process due to the tight integration with the volume service.
+
+## Decision
+
+This standard decides to go with the second option and makes the volume backup feature mandatory in the following way:
+
+In an SCS cloud, the volume backup functionality MUST be configured properly and its API as defined per `/v3/{project_id}/backups` MUST be offered to customers.
+If using Cinder, a suitable [backup driver](https://docs.openstack.org/cinder/latest/configuration/block-storage/backup-drivers.html) MUST be set up.
+
+The volume backup target storage SHOULD be a separate storage system from the one used for volumes themselves.
```

Comment:
Would there be a benefit of using volume backups over snapshots or the simple replication, when it is stored in the same backend?

And how do we address small deployments with only one rack (e.g. in edge cloud scenarios)? Would a second separate storage be reasonable for such deployments?

Reply:
> Would there be a benefit of using volume backups over snapshots or the simple replication, when it is stored in the same backend?

`openstack volume backup restore` has a `--force` flag that enables users to overwrite existing volumes with the backup, which you cannot do with a snapshot afaik. However, there is not much benefit to it since the target volume must be in "available" status and must not be attached to a server even if the server is shutdown. Also, the backups are not bound to the existence of the source volume and the source volume can be deleted while backups being preserved in contrast to snapshots.
I don't think it makes *that* much of a difference in real-world scenarios though.

Reply:
> And how do we address small deployments with only one rack (e.g. in edge cloud scenarios)? Would a second separate storage be reasonable for such deployments?

Good point. I guess this could be a valid reason for this "SHOULD" to be disregarded by a CSP.

Reply:
I still wonder, how a customer can differentiate between a deployment, that has a different store  for the volume backups and a deployment, that does not fulfill this.
Is that easy to determine for a customer?
I don't want users to accidentally think their volume can be backuped, now everything is fine, when volume and backup are stored in the same storage.

Reply:
> I still wonder, how a customer can differentiate between a deployment, that has a different store for the volume backups and a deployment, that does not fulfill this.
> Is that easy to determine for a customer?

Sadly not. This is a `cinder.conf` thing only and not visible to the API user or customer. Either the standard will enforce it so that the guarantee is there if the CSP is compliant or we need some kind of self-description.

> I don't want users to accidentally think their volume can be backuped, now everything is fine, when volume and backup are stored in the same storage.

That's the same feature availability vs. backup reliability topic which came up recently I guess. Added this to the next IaaS call agenda.